$NetBSD: patch-tests_scripts_features_parallelism,v 1.1 2013/05/12 05:47:24 richard Exp $

upstream: http://git.savannah.gnu.org/cgit/make.git/commit/?id=fbe5b2c9cd612b962836b9391fa2b58c8baed6fc
Force intermediate targets to be considered if their non-intermediate

--- tests/scripts/features/parallelism.orig	2010-07-05 18:32:03.000000000 +0000
+++ tests/scripts/features/parallelism
@@ -194,6 +194,23 @@ rm main.x");
     rmfiles(qw(foo.y foo.y.in main.bar));
 }
 
+# Ensure intermediate/secondary files are not pruned incorrectly.
+# See Savannah bug #30653
+
+utouch(-15, 'file2');
+utouch(-10, 'file4');
+utouch(-5, 'file1');
+
+run_make_test(q!
+.INTERMEDIATE: file3
+file4: file3 ; @mv -f $< $@
+file3: file2 ; touch $@
+file2: file1 ; @touch $@
+!,
+	'--no-print-directory -j2', "touch file3");
+
+#rmfiles('file1', 'file2', 'file3', 'file4');
+
 if ($all_tests) {
     # Jobserver FD handling is messed up in some way.
     # Savannah bug #28189
